tests/test_signing.py

```diff
@@ -28,3 +28,9 @@ def test_sign_without_specifying_key(self):
         s = Sign()
 
         assert s.key == 'NCTpkICMlTXie5te9nJniMj9aVbPM6lsjeq5iDZ0dqY='
+
+    def test_sign_incorrect_padding(self):
+        with pytest.raises(InvalidSecretKey):
+            padded_secret_key = "AQAAQDhAAMAAQAAAAAAAAthAAAAJDczODFmZDM2LTNiOTYtNDVmYS04MjQ2LWRkYzJkMmViYjQ2YQ==="
+            s = Sign(padded_secret_key)
+            assert s.sign('value')
```

Comment:
While playing with this i notice that this test didn't pass here that I made to test the new addition:

```python
    def test_sign_incorrect_padding(self):
        with pytest.raises(InvalidSecretKey):
            padded_secret_key = "AQAAQDhAAMAAQYS04MjQ2LWRkYzJkMmViYjQ2YQ==="
            s = Sign(padded_secret_key)
            assert s.sign('value')
```

it was throwing a `ValueError` exception. We can fix this by making something like this in the Sign method:

```python
        except (binascii.Error, ValueError):
            raise InvalidSecretKey("You have passed an invalid secret key of: {}. Make sure you have correctly added your secret key.".format(self.key))
```

this way we catch both exceptions